classes/Search.php

```diff
@@ -0,0 +1,572 @@
+        $qd = $SEARCHENGINE->valid ? $SEARCHENGINE->query_description_short() : $searchstring;
+        $pagetitle = 'Search for ' . $qd;
+        $pagenum = get_http_var('p');
+        if (!is_numeric($pagenum)) {
+            $pagenum = 1;
+        }
+        if ($pagenum > 1) {
+            $pagetitle .= ", page $pagenum";
+        }
+
+        $DATA->set_page_metadata($this_page, 'title', $pagetitle);
+        $DATA->set_page_metadata($this_page, 'rss', 'search/rss/?s=' . urlencode($searchstring));
+        if ($pagenum == 1) {
+            # Allow indexing of first page of search results
+            $DATA->set_page_metadata($this_page, 'robots', '');
+        }
+
+        $o = get_http_var('o');
+        $args = array (
+            's' => $searchstring,
+            'p' => $pagenum,
+            'num' => get_http_var('num'),
+            'pop' => get_http_var('pop'),
+            'o' => ($o=='d' || $o=='r' || $o=='o') ? $o : 'd',
+        );
+
+        $sort_order = 'newest';
+        if ( $o == 'o' ) {
+            $sort_order = 'oldest';
+        } else if ( $o == 'r' ) {
+            $sort_order = 'relevance';
+        }
+
+        $members = null;
+        $cons = null;
+        $glossary = null;
+        if ($pagenum == 1 && $args['s'] && !preg_match('#[a-z]+:[a-z0-9]+#', $args['s'])) {
+            $members = $this->find_members($args['s']);
+            $cons = $this->find_constituency($args);
+            $glossary = $this->find_glossary_items($args);
+        }
+
+        if (!defined('FRONT_END_SEARCH') || !FRONT_END_SEARCH) {
+            return array(
+                'error' =>'Apologies, search has been turned off currently for performance reasons.'
+            );
+        }
+
+        if (!$SEARCHENGINE->valid) {
+            return array('error' => $SEARCHENGINE->error);
+        } else {
+            $LIST = new \HANSARDLIST();
+            $data = $LIST->display('search', $args , 'none');
+            $data['sort_order'] = $sort_order;
+            $data['members'] = $members;
+            $data['cons'] = $cons;
+            $data['glossary'] = $glossary;
+            return $data;
+        }
+    }
+
+    private function search_order_p($searchstring) {
+        global $DATA, $this_page;
+
+        $q_house = '';
+        if (ctype_digit(get_http_var('house'))) {
+            $q_house = get_http_var('house');
+        }
+
+        $wtt = get_http_var('wtt');
+        if ($wtt) {
+            $q_house = 2;
+        }
+
+        # Fetch the results
+        $data = search_by_usage($searchstring, $q_house);
+
+        if ($wtt) {
+            $q_house = 2;
+            $pagetitle = 'League table of Lords who say ' . $data['pagetitle'];
+        } else {
+            $pagetitle = 'Who says ' . $data['pagetitle'] . ' the most?';
+        }
+        $DATA->set_page_metadata($this_page, 'title', $pagetitle);
+
+
+        $data['house'] = $q_house;
+        $data['wtt'] = $wtt;
```

Comment:
Sorry, bit abrupt, I just meant the setting of `$data['wtt']` is done later on in the process when it calls `get_wtt_options`, so no need to do it here.